crates/turborepo-ui/src/log_replayer.rs

```diff
@@ -0,0 +1,93 @@
+use std::{
+    fmt::Display,
+    fs::File,
+    io::{BufRead, BufReader, Write},
+};
+
+use tracing::{debug, warn};
+use turbopath::AbsoluteSystemPath;
+
+use crate::{Error, StyledObject, UI};
+
+#[allow(dead_code)]
+pub struct PrefixedUI<D: Display + Clone, W: Write> {
+    ui: UI,
+    prefix: StyledObject<D>,
+    output: W,
+}
+
+#[allow(dead_code)]
+impl<D: Display + Clone, W: Write> PrefixedUI<D, W> {
+    pub fn new(ui: UI, prefix: StyledObject<D>, output: W) -> Self {
+        Self { ui, prefix, output }
+    }
+
+    pub fn output(&mut self, message: impl Display) -> Result<(), Error> {
+        writeln!(
+            self.output,
+            "{} {}",
```

Comment:
Looking at the Go implementation I see a few differences here that I'm not sure if they were intentional:
 - Go just concats the prefix and the message if the message is non-empty
 - If an empty message is passed then only an empty line will be written to the ui/writer
 - If when replaying the logs there's an empty line then we write `"{prefix}\n"`. This implementation would write `"{prefix} \n"`.

I'd much rather keep this (weird) behavior until we have all the ui/log code ported to avoid our future selves forgetting these differences when we port code that calls these functions.

Reply:
With the second point, we have a workaround in the log replayer code to print empty lines with the prefix. Since I'm writing the prefixed UI implementation, I decided to just have it do what we want and avoid the workaround. Looking at the other uses of prefixed UI, we don't ever print an empty message, so I think this is fine.

As for the other points, I can remove the space